addons/website_profile/static/src/components/profile_dialog/profile_dialog.js

```diff
@@ -0,0 +1,149 @@
+    static components = {
+        ...ConfirmationDialog.components,
+        Wysiwyg,
+    };
+    static props = {
+        ...ConfirmationDialog.props,
+        focusWebsiteDescription: {
+            type: Boolean,
+            optional: true,
+        },
+        userId: { type: Number },
+    };
+    static defaultProps = {
+        ...ConfirmationDialog.defaultProps,
+        cancel: () => {},
+        cancelLabel: _t("Discard"),
+        confirm: () => true,
+        confirmLabel: _t("Update"),
+        dismiss: () => {},
+        focusWebsiteDescription: false,
+        title: _t("Edit Profile"),
+    };
+
+    setup() {
+        super.setup();
+        this.orm = useService("orm");
+        const componentUUID = uuid();
+        this.getUniqueDomID = (suffix) => `profile-dialog-${componentUUID}-${suffix}`;
+        this.upload = useRef("upload");
+        this.profileImg = useRef("profileImg");
+        this.profileImgData = null;
+        this.state = useState({ hasError: false, emailHasError: false, nameHasError: false });
+        useAutofocus({ refName: "name" });
+        const websiteDescriptionClass = this.getUniqueDomID("websiteDescription");
+        onWillStart(async () => {
+            const fields = [
+                "city",
+                "country_id",
+                "email",
+                "name",
+                "website",
+                "website_description",
+                "website_published",
+            ];
+            const [users, countries] = await Promise.all([
+                this.orm.read("res.users", [this.props.userId], fields),
+                this.orm.searchRead("res.country", [], ["id", "name"]),
+            ]);
+            this.user = reactive(users[0], () => this.validate());
+            this.user.country_id = this.user.country_id && this.user.country_id[0]; // keep only id
+            this.countries = countries;
+            const isInternalUser = await user.hasGroup("base.group_user");
+            this.descriptionConfig = {
+                allowFile: isInternalUser,
+                allowImage: isInternalUser,
+                classList: ["form-control", websiteDescriptionClass],
+                content: this.user.website_description,
+                debug: !!this.env.debug,
+                direction: localization.direction || "ltr",
+                placeholder: _t("Write a few words about yourself..."),
+            };
+        });
+        onMounted(() => {
+            if (this.props.focusWebsiteDescription) {
+                setTimeout(() => {
+                    const websiteDescription = document.querySelector(
+                        `.${websiteDescriptionClass}[contenteditable="true"]`
+                    );
+                    if (websiteDescription) {
+                        setCursorEnd(websiteDescription);
+                    }
+                }, 50);
+            }
+            this.validate();
+        });
+    }
+
+    get websiteDescription() {
```

Comment:
Looks only used once, maybe inline code? "websiteDescription" is pretty annoying to grep.